### ISSUE TITLE
Make ibus-ui-gtk3 exits when the name is replaced

### ISSUE THREAD

**Issue by wengxt** [ibus/ibus#1919]
The code specifies the flag "allow replacement". But when it gets replaced, it won't exit but keep running. I don't see much point to keep it running, it won't even try to regain the name if other panel disappears.

If it exits, it would be easier for desktop specific panel to kick in during autostart and replace it. E.g. kimpanel in KDE.

**Comment by fujiwarat** [comment]
> it won't even try to regain the name if other panel disappears.

I don't think so. If the another acquired panel exists, the lost panel gets the owner again in this ibus panel.

**Comment by fujiwarat** [comment]
You can run two /usr/lib*/ibus-ui-gtk3 .

**Comment by wengxt** [comment]
Ok, great, so this is actually better than I thought.

Then the problem is the indicator doesn't hide itself. There's no "Un"RegisterStatusNotifierItem method, I think you should pass "RegisterStatusNotifierItem" with a dbus name, instead of object path, otherwise it won't get notification on the StatusNotifier side.

**Comment by fujiwarat** [comment]
OK, I understood you mean AppIndicator mode now. ibus-ui-gtk3 has two modes of GtkStatusIcon and AppIndicator with XDG_CURRENT_DESKTOP variable and yes it's a bug in the AppIndicator mode.

**Comment by fujiwarat** [comment]
Probably I don't understand your suggestion.
If the argument of g_dbus_connection_register_object() is not object path, the API fails.
If the argument of g_dbus_connection_register_object() and the argument of g_dbus_proxy_call("RegisterStatusNotifierItem".,,)  are different, the icon cannot be launched.

**Comment by wengxt** [comment]
Check the code would be easier: 
https://cgit.kde.org/plasma-workspace.git/tree/statusnotifierwatcher/statusnotifierwatcher.cpp#n58

You may pass either objectpath or service name to RegisterStatusNotifierItem. When it's service name, the path will be "/StatusNotifierItem".

My suggestion is, request a unique service name (maybe org.freedesktop.IBus-<pid>, that's what kde's app is doing). Register an object with path "/StatusNotifierItem".

When you want to hide the indicator, release this service name.
